Tipster/Tipster.h

```diff
@@ -21,20 +21,46 @@ class Tipster : public BTextView
 		void MouseDown(BPoint pt);
 		void MessageReceived(BMessage *msg);
 		void AttachedToWindow();
+		
+		void OpenURL(BString* url);
 	
 	private:
+		struct tLink {
+			const int32 linkoffset;
+			const int32 linklen;
+			const BString target;
+			
+			tLink(int32 off, int32 len, BString trg)
+				:
+				linkoffset(off),
+				linklen(len),
+				target(trg)
+			{
+			}
+		};
+		
 		void UpdateTip(void);
 		void LoadTips(entry_ref ref);
-
+		
 		entry_ref GetTipsFile(void);
+		tLink* GetLinkAt(BPoint point);
+		const char* GetArtworkTitle(BString category);
+		
+		BList links;
+		BList fTipsUsed;
 
 		uint32 fTipNumber;
 		BStringList fTipsList;
+		int32 fTipsLength;
 		
 		BString *fRandomTip;
 		
 		bigtime_t fTime;
 		BMessageRunner *fRunner;
+		
+		BMessenger* messenger;
+		
+		static text_run_array linkStyle;
```

Comment:
Why does the link style need to be static? Do you have multiple instances of the Tipster class? And itt's modified in a non-static member function (AttachedToWindow())...